app/code/Ometria/Core/Service/Product/Inventory.php

```diff
@@ -122,6 +163,49 @@ private function getMSISalableQuantity(ProductInterface $product)
         return $qty;
     }
 
+    /**
+     * @param ProductInterface $product
+     * @return float|null
+     * @throws InputException
+     * @throws LocalizedException
+     * @throws SkuIsNotAssignedToStockException
+     */
+    private function getMSIGlobalSalableQuantity(ProductInterface $product)
+    {
+        /** @var GetAssignedStockIdsBySku $getSalableQtyData */
+        $getAssignedStockIdsBySku = ObjectManager::getInstance()->get(GetAssignedStockIdsBySku::class);
+
+        /** @var GetStockItemConfigurationInterface $getStockItemConfiguration */
+        $getStockItemConfiguration = ObjectManager::getInstance()->get(GetStockItemConfigurationInterface::class);
+
+        /** @var GetProductSalableQtyInterface $getProductSalableQtyInterface */
+        $getProductSalableQty = ObjectManager::getInstance()->get(GetProductSalableQtyInterface::class);
+
+        $qty = 0.;
+        $stockIds = $getAssignedStockIdsBySku->execute($product->getSku());
+
+        foreach ($stockIds as $stockId) {
+            $stockItemConfiguration = $getStockItemConfiguration->execute($product->getSku(), $stockId);
+            $isManageStock = $stockItemConfiguration->isManageStock();
+
+            if (!$isManageStock) {
+                return null;
+            }
+
+            try {
+                // Try to get qty for current stock ID
+                $qtyForStockId = $getProductSalableQty->execute($product->getSku(), $stockId);
+            } catch (InputException $e) {
+                // Catch exception for invalid product types
```

Comment:
shouldn't we at least error log the exception ?
I know that those logs will not go to us

but it may help clients to identify some issues on their side